### PR TITLE
issue #16 Add page to browse/search for named glosses

### PR DESCRIPTION
**Pull request by TheTuTran**
Steps to run:
1. run `npm i` in the directory to install all dependencies. 
2. run `npm run dev` and open up localhost:3000 on browser of choice

- there is an error on the development server where useEffect will run twice because the development server renders the component multiple times. This results in glosses being shown twice in the table, but it should work on production server if you run `npm run build` then `npm run start`.

3. Use the navigation bar or button on home screen to go to the browse glosses page.

New Features:

1. Table with glosses
- Added function to search glosses by name
- Added function to query for all properties with matching targetId
- Added function to clean up the returned properties
- Added classes for glosses and validators using zod
- Mobile responsive, but table looks a little weird when expanding text.
- Users can't change the pagination on the table while the table is still loading glosses
2. Added [slug] page for gloss instance viewing

current state:

image

image

image

### CODE REVIEW

**Comment by AbhilashKotha** [comment]
@TheTuTran It's working well as shown in the demo. 

Is it intentional to have browse by book default when clicking on browse glosses? My suggestion is either show all glosses or have hyperlinks or buttons for browse by book, theme or tag

image

**Comment by AbhilashKotha** [comment]
@TheTuTran all the other things are working fine. We can merge this once my question is answered.

**Review by AbhilashKotha — APPROVED** [review]
looks good to me;

File: src/lib/utils.ts [hunk 1]
```diff
@@ -1,6 +1,134 @@
...
+    textValue: undefined,
+    creator: undefined,
+  };
+
+  processedGloss.targetId = targetId;
+
+  gloss.forEach((item) => {
+    if (!item) return;
+
+    if (item.title && item.title.value) {
```
Comment (thehabes) [inline src/lib/utils.ts:104]: This is a little blocky and probably has a better syntax than a bunch of `if-else` statements.  Think about something like map/filter/reduce

https://medium.com/poka-techblog/simplify-your-javascript-use-map-reduce-and-filter-bd02c593cc2d

**Review by thehabes — APPROVED** [review]
Really start to think about [how you can cache as much information as possible](https://www.geeksforgeeks.org/http-headers-last-modified/).  For example,
imagine having a running, cached entity map like 
`{'entityURI':{prop1:"A", prop2:["B", C"],...}}` 

Act like a `post()` is the last thing you want to do.  I say this because every page refresh seems to ask for the same information again.  This causes some long load times, as well as taxes RERUM unnecessarily.  

RERUM _SHOULD_ have properly configured `Last-Modified` HTTP headers as well as an accurate `__rerum.createdAt` on all objects.

**Pull request merged**